### PR TITLE
337 roles UI update

### LINKED ISSUE #337: Update roles UI
Currently, the roles route shows a list of all roles and the form the create new forms right away. these need to be either split up or made to match the UI mockups. 
### What needs to be done
 
Update the vue to match the mockups. 
If changes are drastic, make sure tests are updated to match.

### PR DESCRIPTION
**Pull request by jonathanp16**
##### _What existing problem does the pull request solve?_
Update the roles UI to match the mockups

image

##### _Demonstrate the code is solid. </br> Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI._

image

##### _List all issues this PR resolves_
- closes #337 

##### _Complete these items before requesting reviews_
- [x] Static Analysis Rules Validated
- [x] All commits follow naming conventions

### Sub-tasks Checklist
- [x] UI Implementation
- [x] UI Test
- [ ] Backend Implementation
- [ ] Backend Tests
- [ ] Integration Tests
- [ ] System Tests
- [x] Documentation

### CODE REVIEW

File: tests/Javascript/Components/Tables/RoleTable.spec.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  InertiaFormMock.post.mockClear()
+
+  localVue = createLocalVue()
+  localVue.use(InertiaApp)
+  localVue.use(InertiaForm)
+
+});
+afterEach(() => {
+  localVue = null
+})
```
Comment (WindOfRussia) [inline tests/Javascript/Components/Tables/RoleTable.spec.js:24]: random q,  do you think its possible to extract this somehow? I can see that the localVue setup is definitely something all tests need.

File: resources/js/Components/Tables/RoleTable.vue [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+          <th class="lt-grey p-3" id="id_role_name">Role Name</th>
+          <th class="lt-grey p-3" id="id_role_set">Permission Set</th>
+          <th class="lt-grey p-3" id="id_role_action">Action</th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr v-for="role in roles" :key="role.id">
+          <td class="text-center lt-grey p-3">{{role.name}}</td>
+          <td class="text-center lt-grey p-3">
+            <Dropdown width="48">
```
Comment (WindOfRussia) [inline resources/js/Components/Tables/RoleTable.vue:16]: Do we want to match the way call other vue components in this file (kebab-case - ex: `<update-role-form>`)

**Review by WindOfRussia — DISMISSED** [review]
Great job making this follow the new mockups. Take a look at the suggestions I left. Approving because this satisfies the requirement.

File: tests/Javascript/Components/Tables/RoleTable.spec.js (hunk 1 above)
Comment (alexstojda) [reply tests/Javascript/Components/Tables/RoleTable.spec.js:24]: No, local ur configuration Vaires from test to test, so it need to be defined every time.

**Review by alexstojda — COMMENTED** [review]
@jonathanp16 could you expand the card down so that there isn't the need to scroll on it. I know we have it like that on other forms but I think we should change it. What do you think?

**Review by EvanDime — COMMENTED** [review]
Love the look. I do agree with Alex's suggestion from above

File: resources/js/Components/Tables/RoleTable.vue (hunk 2 above)
Comment (jonathanp16) [reply resources/js/Components/Tables/RoleTable.vue:16]: Yes will change this

**Comment by jonathanp16** [comment]
> @jonathanp16 could you expand the card down so that there isn't the need to scroll on it. I know we have it like that on other forms but I think we should change it. What do you think?

@alexstojda I think we can expand the card down, would need to do so for the other pages structured like this

**Review by WindOfRussia — APPROVED** [review]
Ah I see you swapped to kebab case and extended the create card all the way down. Looking good.

File: resources/js/Components/Tables/RoleTable.vue (hunk 2 above)
Comment (WindOfRussia) [reply resources/js/Components/Tables/RoleTable.vue]: Closing

File: tests/Javascript/Components/Tables/RoleTable.spec.js (hunk 1 above)
Comment (WindOfRussia) [reply tests/Javascript/Components/Tables/RoleTable.spec.js:24]: Makes sense but we do have repeated code in there. ex: all vue components should load inertia and form. would be nice to have a similar setup to PHPUnit tests where subclasses can replace the beforeEach() method should the need arise.

**Review by KeeganWoods1 — DISMISSED** [review]
LGTM!

**Review by Alex-Hayeur — APPROVED** [review]
Looks good now that the requested changes have been implemented.

**Review by KeeganWoods1 — APPROVED** [review]

**Pull request merged**